src/openforms/registrations/api/views.py

```diff
@@ -56,43 +59,77 @@ def get_objects(self):
+class BaseCatalogiListView(ListMixin, APIView):
+    authentication_classes = (authentication.SessionAuthentication,)
+    permission_classes = (permissions.IsAdminUser,)
+    serializer_class = CatalogusSerializer
+    filter_serializer_class: ClassVar[type[BaseAPIGroupQueryParamsSerializer]]
+
+    def get_objects(self):
+        filter_serializer = self.filter_serializer_class(
+            data=self.request.query_params,
+        )
+        filter_serializer.is_valid(raise_exception=True)
+
+        client = filter_serializer.get_ztc_client()
+        catalogus_data = client.get_all_catalogi()
+        return factory(Catalogus, catalogus_data)
+
+
+class BaseInformatieObjectTypenListView(ListMixin, APIView):
     """
-    List the available InformatieObjectTypen based on the configured registration backend and ZGW APIs services.
+    List the available InformatieObjectTypen.
+
+    Each InformatieObjectType is uniquely identified by its 'omschrijving', 'catalogus',
+    and beginning and end date. If multiple same InformatieObjectTypen exist for different dates,
+    only one entry is returned.
     """
 
     authentication_classes = (authentication.SessionAuthentication,)
     permission_classes = (permissions.IsAdminUser,)
-    serializer_class = InformatieObjectTypeChoiceSerializer
+    serializer_class = InformatieObjectTypeSerializer
+    filter_serializer_class: ClassVar[type[BaseAPIGroupQueryParamsSerializer]]
 
     def get_objects(self):
-        filter_serializer = ListInformatieObjectTypenQueryParamsSerializer(
-            data=self.request.query_params
-        )
+        filter_serializer = self.filter_serializer_class(data=self.request.query_params)
         filter_serializer.is_valid(raise_exception=True)
 
         client = filter_serializer.get_ztc_client()
-        if not client:
-            return []
 
         catalogus_data = client.get_all_catalogi()
         catalogus_mapping = {
             catalogus["url"]: catalogus for catalogus in catalogus_data
         }
 
-        iotypen_data = client.get_all_informatieobjecttypen()
-        iotypen = [
-            {
-                "informatieobjecttype": factory(InformatieObjectType, iotype),
-                "catalogus": factory(Catalogus, catalogus_mapping[iotype["catalogus"]]),
-            }
-            for iotype in iotypen_data
-        ]
+        catalogus = filter_serializer.validated_data.get("catalogus_url", "")
+
+        iotypen_data = client.get_all_informatieobjecttypen(catalogus=catalogus)
+
+        iotypen = []
+
+        for iotype in iotypen_data:
+            # fmt: off
+            exists = any(
+                existing_iotype["catalogus_domein"] == catalogus_mapping[iotype["catalogus"]]["domein"]
+                and existing_iotype["catalogus_rsin"] == catalogus_mapping[iotype["catalogus"]]["rsin"]
+                and existing_iotype["omschrijving"] == iotype["omschrijving"]
+                for existing_iotype in iotypen
+            )
```

Comment:
discussed in meeting, we can probably simplify this check by considering the `(catalogus_url, omschrijving)` unique tuples and only expand the domein/RSIN information after all the data was gathered/de-duplicated.